### PR TITLE
Add Deprecation warning for SPACING

### PR DESCRIPTION
**Pull request by daigof**
- also fixed a map key prop warning in constants story.

### CODE REVIEW

**Comment by snags88** [comment]
Oh actually 1 thing. Could we look at the deprecation warning util and only display that for non-production environment? Not sure how React does it, but it's pretty nice when they only show warnings/deprecation messages in dev

**Comment by daigof** [comment]
> Oh actually 1 thing. Could we look at the deprecation warning util and only display that for non-production environment? Not sure how React does it, but it's pretty nice when they only show warnings/deprecation messages in dev

Added a conditional to display only in test/development environments

File: src/utils/withDeprecationWarning/index.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
+const allowedEnvironments = ['test', 'development', 'dev'];
+const conditional1 = allowedEnvironments.includes(process.env.NODE_ENV);
```
Comment (snags88) [inline src/utils/withDeprecationWarning/index.js]: could we rename this variable to something more descriptive like `shouldShowForEnvironment`

File: src/utils/withDeprecationWarning/index.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+
 export default function withDeprecationWarning(obj, deprecatedProperties = {}) {
   const handler = {
     get(target, property) {
-      if (Object.keys(deprecatedProperties).includes(property)) {
-        console.warn(`[Deprecation Warning]: ${deprecatedProperties[property]}`);
+      const conditional2 = Object.keys(deprecatedProperties).includes(property);
```
Comment (snags88) [inline src/utils/withDeprecationWarning/index.js]: also should update this variable name to `isDeprecatedProperty`

**Pull request merged**